esmvaltool/diag_scripts/extreme_events/ExtremeEvents.r

```diff
@@ -0,0 +1,357 @@
+# #############################################################################
+# ExtremeEvents.r
+#
+# Authors: Björn Brötz (DLR, Germany)
+#          Marit Sandstad (CICERO, Norway)
+#          Christian W. Mohr (CICERO, Norway)
+# #############################################################################
+# Description
+#    Calculate extreme events with plotting functionality
+#
+# Modification history
+#    2019 0506-hard_jo  : conversion to ESMValTool2
+#    2018 1006-A_cwmohr : observation read and sorting fixes
+#    2018 1003-A_cwmohr : correcting r.interface output for observation data.
+#    2018 0725-A_cwmohr : modification of timeseries_main() and climdex selection
+#    2018 0615-A_cwmohr : more clean up of code
+#    2018 0131-A_laue_ax: clean-up of code, adaptation to ESMValTool standards
+#                        added tagging support
+#    2017 0920-A_sand_ma: modification to include plotting
+#    2016 0414-A_broe_bj: written
+# ############################################################################
+
+library(climdex.pcic.ncdf)  # nolint
+library(tools)
+library(yaml)
+library(ncdf4)
+library(ncdf4.helpers)
+
+cdo <- function(command, args = "", input = "", options = "", output = "",
```

Comment:
Would it be possible to move this function to esmvaltool/diag_scripts/shared and use it from there in the diagnostics that use it, instead of having a copy everywhere?

Reply:
Good idea, I just implemented it. There is now a `diag_scripts/shared/external.R` file containing the two cdo and nco external wrappers. We can later then make adjustments also to the other  metrics which use them, once this PR is completed.

PS: I also renamed all .r files to .R since this seems to be generally considered better style and it is what we used in the other R metrics. I also avoid the CamelCase and `ExtremeEvents.r `is now e`xtreme_events.R`.

Reply:
Thanks

Reply:
It would be nice if the same thing could be done for the code that reads the settings.yml file and metadata.yml files, because that's also copy/pasted in every R diagnostic, but if it's too much work now we can leave it for later.